src/auth.rs

```diff
@@ -108,31 +101,22 @@ impl Authenticator {
                                                       String,
                                                       u32);
 
-        let rnonce_b64 = match rnonce_opt {
-            Some(val) => val,
-            None => return Err(ResponseError(String::from("Invalid rnonce returned"))),
-        };
+        let rnonce_b64 = rnonce_opt
+            .ok_or(ResponseError(String::from("Invalid rnonce returned")))?;
```

Comment:
I wonder if it might be worth figuring out if there are any clippy lints we don't care about, fixing all the others, and then requiring that clippy passes on any future PR (which we could potentially do by just having clippy run as part of the Travis build). Not sure how feasible that would actually be, though...